drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -917,23 +918,15 @@ int stm32_clock_control_init(const struct device *dev)
 	optimize_regulator_voltage_scale(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
 
 	z_stm32_hsem_unlock(CFG_HW_RCC_SEMID);
-
-	/* Update CMSIS variable */
-	SystemCoreClock = CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC;
-
-	return r;
-}
 #else
-int stm32_clock_control_init(const struct device *dev)
-{
 	ARG_UNUSED(dev);
```

Comment:
Even this line could be common code, and then there is no need for a #else.

Reply:
> Even this line could be common code, and then there is no need for a #else.

Please put a code-suggestion here. Then I will take it over and you become co-author of this PR. ;)

Reply:
Thanks, but there is no need for that, I'm just the reviewer here, not the author :)

Reply:
@gautierg-st Good catch. Thank you. It is covered now by d4fad869ff02b5b0724f54c04977c89cad136df0

Reply:
Thanks, could you just squash your two commits into a single one, since it is part of the same change?

Reply:
> Thanks, could you just squash your two commits into a single one, since it is part of the same change?

Squashed as requested.